src/components/client.ts

```diff
@@ -63,8 +72,18 @@ export class <%= className %> {
         break;
     }
     this.rpc = new Client(new RequestManager([this.transport]));
-    this.validator = new MethodCallValidator(<%= className %>.openrpcDocument);
   }
+
+  /**
+   * Adds a JSONRPC notification handler to handle receiving notifications.
+   * @example
+   * myClient.onNotification((data)=>console.log(data));
+   */
+  public async initialize() {
+    this.dereffedDocument = await parseOpenRPCDocument(<%= className %>.openrpcDocument);
```

Comment:
could do this in constructor
```suggestion
    this.dereffedDocumentPromise = parseOpenRPCDocument(<%= className %>.openrpcDocument);
```

and then in `request`

```
await this.dereffedDocumentPromise;
```

Reply:
done thanks!